File: source/GyeongHo/boj/2447.cpp
#include <iostream>
using namespace std;

void star(int i, int j, int n)
{
    if ((i / n) % 3 == 1 && (j / n) % 3 == 1)
        cout << ' ';
    else if (n == 1)
        cout << '*';
    else
        star(i, j, n / 3);
}

int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            star(i, j, n);
        }
        cout << '\n';
    }
    return 0;
}

File: source/GyeongHo/boj/2447.py
import sys
input = sys.stdin.readline
print = sys.stdout.write

def star(i, j, n):
    if ((i // n) % 3 == 1 and (j // n) % 3 == 1):
        print(' ')
    elif (n == 1):
        print('*')
    else:
        star(i, j, n // 3)

n = int(input())
for i in range(n):
    for j in range(n):
        star(i, j, n)
    print('\n')